pkg/scheduler/framework/runtime/framework.go

```diff
@@ -397,9 +397,9 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 			if status.IsUnschedulable() {
 				return status
 			}
-			msg := fmt.Sprintf("prefilter plugin %q failed for pod %q: %v", pl.Name(), pod.Name, status.Message())
-			klog.Error(msg)
-			return framework.NewStatus(framework.Error, msg)
+			err := status.AsError()
+			klog.ErrorS(err, "Failed running PreFilter plugin", "plugin", pl.Name(), "pod", klog.KObj(pod))
+			return framework.AsStatus(fmt.Errorf("prefilter plugin %q failed for pod \"%v/%v\": %w", pl.Name(), pod.Namespace, pod.Name, err))
```

Comment:
what's the point of coming up different error message for `klog` and `status`?

Reply:
The klog error is to be displayed immediately, thus, it should have all the information necessary for debugging.

The status is to be returned, and will be eventually displayed in a k8s event (like when you do `kubectl describe pod`). And this error might be concatenated with more context as it moves up.

Fix now: Do not include the pod name in the status, as it will be redundant up the stack.

For debate: Is this the correct place to produce the klog? My thinking is that, the closer to the actual error is, the easier it is to debug. But then we have some repeated code. The alternative is to klog one level above (in scheduler.go). But then the errors wouldn't show in other binaries, such as cluster autoscaler or the descheduler. Thoughts @damemi @MaciekPytel ?